packages/semcom-sdk/lib/component/services/register-component.service.ts

```diff
@@ -23,7 +23,7 @@ export class RegisterComponentService extends AbstractRegisterComponentService {
 
     }
 
-    const tag = `semcom-${ componentMetadata.tag }-${ btoa(Date.now().toString()).replace('==', '').toLowerCase() }`;
+    const tag = `semcom-${ componentMetadata.tag }-${ btoa(componentMetadata.uri).replace('=', '').toLowerCase() }`;
```

Comment:
This is definitely not enough. The whole point about using an encoded date string is that it is a quickly accessible _SHORT_ addition to the tag. We don't want the resulting HTML to be unreadable. We'll have to find another solution for this, if we want to use the URL as part of it somehow.

Reply:
we could map known components to an index, and increment the index everytime a new component is found, then tag would be `${ componentMetadata.tag }-${ index }`

a substring of a hashing function is also possible